internal/initwd/module_install.go

```diff
@@ -254,6 +276,32 @@ func (i *ModuleInstaller) moduleInstallWalker(ctx context.Context, manifest mods
 					}
 
 					log.Printf("[TRACE] ModuleInstaller: Module installer: %s %s already installed in %s", key, record.Version, record.Dir)
+
+					// Checking for module deprecations in the case no new module versions need installation
+					if addr, isRegistryModule := req.SourceAddr.(addrs.ModuleSourceRegistry); isRegistryModule {
+						regClient := i.reg
+
+						regsrcAddr := regsrc.ModuleFromRegistryPackageAddr(addr.Package)
+						resp, err := regClient.ModuleVersions(ctx, regsrcAddr)
+						if err != nil {
+							log.Printf("[WARN] Deprecation for %s could not be checked: call to registry failed: %v", addr.Package.Namespace, err)
```

Comment:
You may have missed my previous feedback about the `Namespace` of the module not being enough information?

Reply:
Oh whoops! Think my brain squashed your suggestion with the note on how to include it in the diag itself.

Made an edit to the commit to include https://github.com/hashicorp/terraform/pull/35231/commits/7dd73600076f942118a48a8fbe215aee515708ee